src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { FormEvent, useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+type GetUserById = (todoId: number) => User | null;
+
+const getUserById: GetUserById = (todoId: number) => {
+  return usersFromServer.find(user => user.id === todoId) || null;
+};
+
+const getAllIds = (array: Todo[]) => {
+  const ids: number[] = [];
+
+  array.forEach(element => ids.push(element.id));
+
+  return ids;
+};
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(todosFromServer);
+
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const preparedTodos = todoList.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
+
+  const allIds = getAllIds(preparedTodos);
+
+  const hasNoValuesInForm = !title || user === 0;
+
+  const handleReset = () => {
+    setTitle('');
+    setUser(0);
+    setTitleError(false);
+    setUserError(false);
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (user === 0) {
+      setUserError(true);
+    }
```

Comment:
use the same syntax in all such cases
```suggestion
    if (!title) {
      setTitleError(true);
    }

    if (!user) {
      setUserError(true);
    }
```